src/test/ui/closure-move-sync.stderr

```diff
@@ -0,0 +1,24 @@
+error[E0277]: `std::sync::mpsc::Receiver<()>` cannot be shared between threads safely
+  --> $DIR/closure-move-sync.rs:16:13
+   |
+16 |     let t = thread::spawn(|| {
+   |             ^^^^^^^^^^^^^ `std::sync::mpsc::Receiver<()>` cannot be shared safely, consider marking the closure `move`
```

Comment:
the message happens to work here, but I imagine you could also have a scenario like this

```
fn gimme<T: Send>(_: T) { }

fn main() {
    let (send, _recv) = ::std::sync::mpsc::channel();
    gimme(&send);
}
```

What gets printed here?

Reply:
You'll get the label mentioning closures. That's why the original label was more ambiguous. I'll have to expand `rustc_on_unimplemented` to be able to tell if a closure is involved in order to a message that is always correct.

I think that I'd prefer to merge this PR with the previous ambiguous wording ("`std::sync::mpsc::Receiver<()>` cannot be shared safely, if using a closure consider marking it `move`") and work on a follow up to improve that. I'm also waiting on #47613 to land in beta so that this message can be part of a note instead of the label.

Reply:
I could live with an ambiguous message for now, I suppose.

Reply:
Could we file a follow-up bug though describing the ambiguous message and thoughts for improvement? (And tag the message with `// FIXME` and the issue number...)

Reply:
Filed #48534.

Reply:
Updated.